src/clients/cargo.js

```diff
@@ -38,15 +54,32 @@ class Cargo {
   }
 
   async projectMetadata() {
-    const { packages: [{ name, version, metadata },] = [] } = await this.metadata();
-    return { id: `${name} ${version}`, name, version, metadata };
+    const { packages: [{ name, version },] = [] } = await this.metadata();
+    if (_.isEmpty(name) || _.isEmpty(version)) {
+      throw new Error('Failed to resolve project metadata');
+    }
+
+    const id = `${name} ${version}`;
+
+    const wasmFileName = `${name.replace(/-/g, '_')}.wasm`;
+    const wasm = {
+      fileName: wasmFileName,
+      filePath: path.join('target', Cargo.WasmTarget, 'release', wasmFileName),
+      optimizedFilePath: path.join('artifacts', wasmFileName)
+    };
+
+    return { id, name, version, wasm };
   }
 
+  /**
+   * @deprecated since v1.2.0
+   */
```

Comment:
are we documenting this somewhere? And/or adding a log to explain how to avoid this after deprecating it

Reply:
We are not keeping track of those yet. I added it as a reminder to make it easier to spot code that needs an upgrade when using the IDE.